### ISSUE TITLE
SQLPro for SQLite - Data column widths, Filters, data editing

### ISSUE THREAD

**Issue by jmadren** [hankinsoft/SQLPro#721]
The last update(s) have changed many things. Here's the ones I'm having trouble with so far:

Table column widths in Data view - it's not consistent, so it's hard to tell what's going on. On some tables it seems to be auto-fitting to the data (and not the header) for some columns but not others. But this auto-fitting is not always correct (either too narrow or too wide). Then for other tables it doesn't seem to be doing this at all. And for some tables, it doesn't remember the width of some columns after I resize them, but on others it does. And then on some of these, it doesn't return to the original (auto-fitted?) size it had before, but also not to the size I set. It winds up somewhere in between. In all these cases I can see it resizing after opening the table.

Filters - if I set a filter on a Data view, then switch to another table, it seems to be trying to apply the filter that table also. And if I go back to the first table and activate the filter function, it doesn't show the existing filter, but it is obviously still filtering it. Clearing the filter does work and restores the table to the full view.

Data entry - I used to be able to edit a cell, then press the up or down arrow to move to the same cell in the next or previous row, thereby allowing easier data entry/editing on a column of data. Now I have to press Enter to "save" the edit, then press Up or Down, then press Enter again to activate edit mode. Is there an intended reason why this changed? The old way was more convenient.

**Comment by hankinsoft** [comment]
> Data entry

Fixed for the next build:
![CleanShot 2022-01-11 at 09 39 39](https://user-images.githubusercontent.com/3160528/148953199-9d151f4a-c48b-4e25-9154-1abdd3deffc0.gif)

(Also FYI, CMD-Enter works as a quick shortcut for apply without preview).

Still working on the other two items.

**Comment by hankinsoft** [comment]
Filter issue is fixed for the next build. Just trying to figure out the column sizing issue, then I'll submit the build to Setapp.

**Comment by jmadren** [comment]
Got the update. Everything is working as expected. Thanks!

One further enhancement would be nice: When auto adjusting a column's width by double-clicking on the divider in the header, could it consider the header text as part of the "data" so that it doesn't get truncated for narrow columns (like a column of small integers). Otherwise, I have to manually adjust each column anyway to keep the header text visible.

**Comment by hankinsoft** [comment]
> When auto adjusting a column's width......

Sure! I'll get that in for the next build. I'll probably wait a few days before updating it incase anyone else reports any issues with the recent data grid changes.

**Comment by jmadren** [comment]
Cool. One more thing, as I mentioned before, I can see the table/columns briefly display at some default, then adjust to the final picture. I don't know what control you are using, but you should be able to effectively "freeze" the display until the column widths are set. The shifting/adjusting can be a little jarring and distracting.

Just curious, were you trying to automatically auto-fit all columns in the earlier builds? Some other tools do that, with a maximum per column based on some ratio of the current table width. That would be nice if you were and could get it working correctly.

Thanks again for your products and for your prompt response to issues!

**Comment by hankinsoft** [comment]
> you should be able to effectively "freeze" the display until the column widths are set. The shifting/adjusting can be a little jarring and distracting.

I've got code in there to try and prevent this, but I've never been able to truly figure it out. I agree that its jaring.

> Just curious, were you trying to automatically auto-fit all columns in the earlier builds? Some other tools do that, with a maximum per column based on some ratio of the current table width. That would be nice if you were and could get it working correctly.

So with the previous builds, I removed the control used for the data tab completely and replaced it with the results grid you see when running a query. The grid in the data tab was made when I very first started the product ~8 years ago and has only received minor updated to 'keep it working', whereas the results grid is the one used in my main product SQLPro Studio and has more features and overall better code -- I tried to keep most of the features/behaviour from the old data tab such as remember column sizes, but I had a spot where they were being stored at the wrong point which is why you were seeing the odd behaviour.

The data grid tries to size content to the windows width, but if it has too many columns, will instead make them uniform width. The query results grid will actually try and resize to the content.
